src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+            leftHandActor->world.pos.z = 0.5f * (this->player.bodyPartsPos[15].z + this->player.leftHandWorld.pos.z);
+        }
+    } else if (limbIndex == 11) {
+        Actor* actor730 = this->player.unk_730;
+
+        if ((*dList1 != NULL) && this->player.currentMask && !(this->player.stateFlags2 & PLAYER_STATE2_1000000)) {
+            if ((this->player.skelAnime.animation != &gameplay_keep_Linkanim_00D0A8) &&
+                ((this->player.skelAnime.animation != &gameplay_keep_Linkanim_00D0C8) ||
+                 (this->player.skelAnime.curFrame >= 12.0f))) {
+                if (func_80127438(play, &this->player, this->player.currentMask)) {
+                    OPEN_DISPS(play->state.gfxCtx);
+                    gSPDisplayList(POLY_OPA_DISP++, object_mask_ki_tan_DL_0004A0);
+                    CLOSE_DISPS(play->state.gfxCtx);
+                }
+            }
+        }
+        if ((actor730 != NULL) && (actor730->id == ACTOR_BG_IKNV_OBJ)) {
+            Math_Vec3f_Copy(&this->player.actor.focus.pos, &actor730->focus.pos);
+        } else {
+            static Vec3f D_80A418CC = { 1100.0f, -700.0f, 0.0f };
+
+            Matrix_MultVec3f(&D_80A418CC, &this->player.actor.focus.pos);
+        }
+    } else if (limbIndex == 21) {
+        if (D_80A41D60 || (gSaveContext.save.weekEventReg[50] & 0x80) ||
+            (INV_CONTENT(ITEM_PENDANT_OF_MEMORIES) == ITEM_PENDANT_OF_MEMORIES) ||
+            (this->player.unk_B2A - 1 == GID_PENDANT_OF_MEMORIES)) {
+            D_80A41D60 = true;
+        } else {
+            OPEN_DISPS(play->state.gfxCtx);
+            gSPDisplayList(POLY_OPA_DISP++, object_test3_DL_00CB60);
+            CLOSE_DISPS(play->state.gfxCtx);
+        }
+    } else {
+        func_80128B74(play, &this->player, limbIndex);
+    }
+}
+
+static TexturePtr sEyeTextures[] = {
+    object_test3_Tex_000DC0, object_test3_Tex_003680, object_test3_Tex_003E80, object_test3_Tex_004680,
+    object_test3_Tex_004E80, object_test3_Tex_005680, object_test3_Tex_005E80, object_test3_Tex_006680,
+};
+
+static TexturePtr sMouthTextures[] = {
+    object_test3_Tex_0009C0,
+    object_test3_Tex_006E80,
+    object_test3_Tex_007280,
+    object_test3_Tex_007680,
+};
+
+typedef struct {
+    /* 0x0 */ u8 eyeIndex;
+    /* 0x1 */ u8 mouthIndex;
+} KafeiFace; // size = 0x2
+
+static KafeiFace sFaceExpressions[] = {
+    { 0, 0 }, { 1, 0 }, { 2, 0 }, { 0, 0 }, { 1, 0 }, { 2, 0 }, { 4, 0 }, { 5, 1 }, { 7, 2 }, { 0, 2 },
+    { 3, 0 }, { 4, 0 }, { 2, 2 }, { 1, 1 }, { 0, 2 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A409D4.s")
+// static u8 sFaceExpressions[][2] = {
+//     { 0, 0 }, { 1, 0 }, { 2, 0 }, { 0, 0 }, { 1, 0 }, { 2, 0 }, { 4, 0 }, { 5, 1 }, { 7, 2 }, { 0, 2 },
+//     { 3, 0 }, { 4, 0 }, { 2, 2 }, { 1, 1 }, { 0, 2 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 },
+// };
```

Comment:
This can go, as it's already above

Reply:
Oh right, I was unsure whether a struct or a 2D array is preferred. It matches as either.

Reply:
Personally I just say 2d array.

Reply:
Oh, I didn't notice it was a struct vs. 2D array. I actually like the struct a little more, as a `.eyeIndex` and `.mouthIndex` is clearer than `[0]` and `[1]`

Reply:
Oh I do like those better. Nevermind my comment about 2d array :)